include/ginkgo/core/solver/direct.hpp

```diff
@@ -0,0 +1,143 @@
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#ifndef GKO_PUBLIC_CORE_SOLVER_DIRECT_HPP_
+#define GKO_PUBLIC_CORE_SOLVER_DIRECT_HPP_
+
+
+#include <ginkgo/core/base/lin_op.hpp>
+#include <ginkgo/core/factorization/factorization.hpp>
+#include <ginkgo/core/solver/lower_trs.hpp>
+#include <ginkgo/core/solver/solver_base.hpp>
+#include <ginkgo/core/solver/upper_trs.hpp>
+
+
+namespace gko {
+namespace experimental {
+namespace solver {
+
+
+template <typename ValueType, typename IndexType>
+class Direct : public EnableLinOp<Direct<ValueType, IndexType>>,
+               public gko::solver::EnableSolverBase<
+                   Direct<ValueType, IndexType>,
+                   factorization::Factorization<ValueType, IndexType>>,
+               public Transposable {
+    friend class EnablePolymorphicObject<Direct, LinOp>;
+
+public:
+    using value_type = ValueType;
+    using index_type = IndexType;
+    using factorization_type =
+        factorization::Factorization<value_type, index_type>;
+    using transposed_type = Direct;
+
+    std::unique_ptr<LinOp> transpose() const override;
+
+    std::unique_ptr<LinOp> conj_transpose() const override;
+
+    GKO_CREATE_FACTORY_PARAMETERS(parameters, Factory)
+    {
+        std::shared_ptr<const LinOpFactory> factorization;
+
+        parameters_type& with_factorization(
+            std::shared_ptr<const LinOpFactory> factory)
+        {
+            this->factorization = std::move(factory);
+            return *this;
+        }
```

Comment:
TBH, I just wanted to write out these parameters, since the GKO_FACTORY_PARAMETER setup unnecessarily obscures a simple piece of code (with added complexity of perfect forwarding, which is unnecessary in most cases IMO).

Reply:
For reference: #1162